src/metabase/api/card.clj

```diff
@@ -770,7 +757,9 @@ saved later when it is ready."
 
   (let [card (t2/select-one Card :id id)]
     (delete-alerts-if-needed! card-before-update card)
-    (publish-card-update! card archived)
+    (when-not (set/subset? (set (keys card-updates))
```

Comment:
Why do set comparison instead of plain `(= [:archived] (keys card-updates))`? Because I discovered this when I was working on no longer firing `:event/card-update` when we change a card's position. I separated this into a different PR for clarity, but the next step will just be adding `:collection_position` to the set of keys to ignore here.